packages/cryptoassets/src/chains/BaseChain.ts

```diff
@@ -0,0 +1,108 @@
+import { ensure0x } from '@chainify/utils';
+import { getAddress, isAddress } from '@ethersproject/address';
+import { IAsset } from '../interfaces/IAsset';
+import { IChain } from '../interfaces/IChain';
+import { IFees } from '../interfaces/IFees';
+import { IGasLimits } from '../interfaces/IGasLimit';
+import { INetwork } from '../interfaces/INetwork';
+import { ChainId } from '../types';
+import { ExplorerView, FeeMultiplier } from './types';
+
+export abstract class BaseChain implements IChain {
+  constructor(chain: {
```

Comment:
I think you could skip a bunch of lines by making `IChain` only have the properties 
Then accepting IChain in the constructor and doing `Object.assign(this, chain)`

Not sure if that will work for everything else you are doing

Reply:
Good idea